library/spdm_requester_lib/libspdm_req_get_measurement_extension_log.c

```diff
@@ -0,0 +1,352 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2024 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+#include "internal/libspdm_requester_lib.h"
+
+#if LIBSPDM_ENABLE_CAPABILITY_MEAS_CAP
+
+#pragma pack(1)
+typedef struct {
+    spdm_message_header_t header;
+    uint16_t portion_length;
+    uint16_t remainder_length;
+    uint8_t measure_exten_log[LIBSPDM_MAX_MEL_BLOCK_LEN];
+} libspdm_measurement_extension_log_response_max_t;
+#pragma pack()
+
+/**
+ * This function sends GET_MEASUREMENT_EXTENSION_LOG and receives MEASUREMENT_EXTENSION_LOG.
+ *
+ * @param  spdm_context               A pointer to the SPDM context.
+ * @param  session_id                 Indicates if it is a secured message protected via SPDM session.
+ *                                    If session_id is NULL, it is a normal message.
+ *                                    If session_id is not NULL, it is a secured message.
+ * @param  measurement_record         A pointer to a destination buffer to store the HEM measurement record.
+ * @param  measurement_record_length  Indicate the size in bytes of the HEM measurement record.
+ * @param  length                     The len of get MEL in every time.
+ * @param  mel_size                   On input, indicate the size in bytes of the destination buffer to store.
+ *                                    On output, indicate the size in bytes of the MEL.
+ * @param  measure_exten_log          A pointer to a destination buffer to store the MEL.
+ *
+ **/
+static libspdm_return_t libspdm_try_get_measurement_extension_log(libspdm_context_t *spdm_context,
+                                                                  const uint32_t *session_id,
+                                                                  void *measurement_record,
+                                                                  uint32_t measurement_record_length,
+                                                                  uint16_t length,
+                                                                  size_t *mel_size,
+                                                                  void *measure_exten_log)
+{
+    libspdm_return_t status;
+    spdm_get_measurement_extension_log_request_t *spdm_request;
+    size_t spdm_request_size;
+    libspdm_measurement_extension_log_response_max_t *spdm_response;
+    size_t spdm_response_size;
+    uint16_t total_responder_mel_buffer_length;
+    size_t mel_capacity;
+    size_t mel_size_internal;
+    uint16_t remainder_length;
+    uint8_t *message;
+    size_t message_size;
+    size_t transport_header_size;
+    libspdm_session_info_t *session_info;
+    libspdm_session_state_t session_state;
+    bool chunk_enabled;
+    spdm_measurement_block_dmtf_t *spdm_hem;
+    size_t hash_size;
+
+    /* -=[Check Parameters Phase]=- */
+    LIBSPDM_ASSERT(mel_size != NULL);
+    LIBSPDM_ASSERT(*mel_size > 0);
```

Comment:
Can remove this check. The requested size can be any value, including `0`. `0` would be used to determine the size of the MEL without actually fetching it from the Responder.